### PR TITLE
Api provides an endpoint for a products index

### PR DESCRIPTION
**Pull request by designerofthing**
[PT Link](https://www.pivotaltracker.com/story/show/172642004)
```As a restaurant API
In order to serve a client with products
I would like to be able to provide an endpoint for a products index
```
- Wrote Model spec
- Generated Product model
- Wrote request spec
- Added Products controller with index action
- Configured routes for products controller

### CODE REVIEW

**Review by jysmys — APPROVED** [review]
Looks great! good work

File: app/controllers/api/v1/products_controller.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Api::V1::ProductsController < ApplicationController
+  def index
+    products = Product.all
+    render json: { products: products }
+  end
+end
+
```
Comment (oliverochman) [inline app/controllers/api/v1/products_controller.rb]: remove empty lines

File: app/models/product.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Product < ApplicationRecord
+  validates_presence_of :title
```
Comment (oliverochman) [inline app/models/product.rb]: this can be refactored to:
`validates_presence_of :title, :description, :price`

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/models/product.rb (hunk 2 above)
Comment (designerofthing) [reply app/models/product.rb]: done

File: app/controllers/api/v1/products_controller.rb (hunk 1 above)
Comment (designerofthing) [reply app/controllers/api/v1/products_controller.rb]: done

**Pull request merged**